tests/Tests/ORM/Functional/Mapping/PostgreSqlDefaultQuoteStrategyTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Mapping;
+
+use Doctrine\DBAL\Platforms\PostgreSQLPlatform;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\ORM\Mapping\Driver\AttributeDriver;
+use Doctrine\ORM\Tools\SchemaTool;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+use function json_encode;
+use function sprintf;
+
+class PostgreSqlDefaultQuoteStrategyTest extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $platform = $this->_em->getConnection()->getDatabasePlatform();
+        if (! $platform instanceof PostgreSQLPlatform) {
+            self::markTestSkipped(self::class . ' requires the use of postgresql.');
+        }
+
+        $this->_em         = $this->getEntityManager(
+            null,
+            new AttributeDriver([__DIR__], true)
+        );
+        $this->_schemaTool = new SchemaTool($this->_em);
+    }
+
+    public function testQuotingStillAllowsInterpretingDotAsSchemaSeparator(): void
+    {
+        $this->createSchemaForModels(Product::class);
+        $schemaManager = $this->createSchemaManager();
+        self::assertContains(
+            'Inventory',
+            $schemaManager->listSchemaNames(),
+            sprintf(
+                '"Inventory" should be interpreted as a schema name, even when quoted. List of schema names: %s',
+                json_encode($schemaManager->listSchemaNames())
+            )
+        );
+        self::assertContains(
+            'Products',
+            $schemaManager->listTableNames(),
+            '"Products" should be interpreted as a table name, even when quoted.'
+        );
+    }
+}
+
+#[ORM\Entity]
+#[ORM\Table(name: '`Inventory`.`Products`')] // Using backticks to make sure capitalization is preserved
```

Comment:
@morozov I thought this syntax would result in case being preserved, but apparently it results in `inventory` being created. Am I doing something wrong? Note that I'm targeting ORM 2 here, which means DBAL 3 is in use.

Reply:
The schema DDL in this case looks like this:
```sql
CREATE SCHEMA Inventory;
CREATE TABLE "Inventory"."Products" (id INT NOT NULL, PRIMARY KEY(id));
```

The reason is that `Schema::_addTable()`, when adding the namespace of the table being added to the list of namespaces, ignores the fact that the table name is quoted and eventually doesn't quote the name in the `CREATE SCHEMA` statement.

Reply:
So, is that a DBAL bug then? I'll push a version where uppercase letters are only contained in the table name, hopefully that works.

Thanks for looking into this.